py-polars/polars/dataframe/frame.py

```diff
@@ -7586,13 +7586,14 @@ def explode(
         """
         return self.lazy().explode(columns, *more_columns).collect(_eager=True)
 
+    @deprecate_renamed_parameter("columns", "on", version="1.0.0")
     def pivot(
         self,
-        index: ColumnNameOrSelector | Sequence[ColumnNameOrSelector] | None,
-        columns: ColumnNameOrSelector | Sequence[ColumnNameOrSelector] | None,
+        on: ColumnNameOrSelector | Sequence[ColumnNameOrSelector],
+        *,
+        index: ColumnNameOrSelector | Sequence[ColumnNameOrSelector],
```

Comment:
I'm removing the `|None` type hint, as it's not currently supported (will leave that to a separate PR)